diff --git a/arch/arm/boot/dts/qcom/sdm636_e7s.dtsi b/arch/arm/boot/dts/qcom/sdm636_e7s.dtsi
index b7479eea6b43..829a77c0cb50 100644
--- a/arch/arm/boot/dts/qcom/sdm636_e7s.dtsi
+++ b/arch/arm/boot/dts/qcom/sdm636_e7s.dtsi
@@ -42,22 +42,6 @@
 		};
 	};
 
-	devfreq_memlat_4: qcom,arm-memlat-mon-4 {
-		qcom,core-dev-table =
-		< 1113600 762 >,
-		< 1401600 3879 >,
-		< 1804800 5163 >;
-	};
-
-	devfreq_cpufreq: devfreq-cpufreq {
-		mincpubw-cpufreq {
-			cpu-to-dev-map-4 =
-				< 1113600 762 >,
-				< 1401600 2086 >,
-				< 1747200 2929 >,
-				< 1804800 5163 >;
-		};
-	};
 };
 
 &soc {
diff --git a/arch/arm/boot/dts/qcom/sdm660-regulator.dtsi b/arch/arm/boot/dts/qcom/sdm660-regulator.dtsi
index e2d8179af62a..e57c2e39b369 100644
--- a/arch/arm/boot/dts/qcom/sdm660-regulator.dtsi
+++ b/arch/arm/boot/dts/qcom/sdm660-regulator.dtsi
@@ -768,7 +768,7 @@
 					/* Speed bin 3 */
 					<300000000  633600000  902400000
 					1113600000 1401600000 1536000000
-					1612800000 1843200000>,
+					1747200000 1843200000>,
 
 					/* Speed bin 4 */
 					<300000000  633600000  902400000
@@ -918,7 +918,7 @@
 
 					/* Speed bin 3 */
 					<300000000  1113600000 1401600000
-					 1747200000 1804800000 2150400000
+					 1747200000 1958400000 2150400000
 					 2208000000>,
 
 					/* Speed bin 4 */
diff --git a/arch/arm/boot/dts/qcom/sdm660_e7s.dtsi b/arch/arm/boot/dts/qcom/sdm660_e7s.dtsi
index 080469bc5208..54ef5f5a5d5c 100644
--- a/arch/arm/boot/dts/qcom/sdm660_e7s.dtsi
+++ b/arch/arm/boot/dts/qcom/sdm660_e7s.dtsi
@@ -51,7 +51,7 @@
 			device_type = "cpu";
 			compatible = "arm,kryo";
 			reg = <0x0 0x0>;
-			clock-frequency = <1612800000>;
+			clock-frequency = <1843200000>;
 			enable-method = "psci";
 			qcom,limits-info = <&mitigation_profile0>;
 			qcom,lmh-dcvs = <&lmh_dcvs0>;
@@ -82,7 +82,7 @@
 			device_type = "cpu";
 			compatible = "arm,kryo";
 			reg = <0x0 0x1>;
-			clock-frequency = <1612800000>;
+			clock-frequency = <1843200000>;
 			enable-method = "psci";
 			qcom,limits-info = <&mitigation_profile0>;
 			qcom,lmh-dcvs = <&lmh_dcvs0>;
@@ -107,7 +107,7 @@
 			device_type = "cpu";
 			compatible = "arm,kryo";
 			reg = <0x0 0x2>;
-			clock-frequency = <1612800000>;
+			clock-frequency = <1843200000>;
 			enable-method = "psci";
 			qcom,limits-info = <&mitigation_profile0>;
 			qcom,lmh-dcvs = <&lmh_dcvs0>;
@@ -132,7 +132,7 @@
 			device_type = "cpu";
 			compatible = "arm,kryo";
 			reg = <0x0 0x3>;
-			clock-frequency = <1612800000>;
+			clock-frequency = <1843200000>;
 			enable-method = "psci";
 			qcom,limits-info = <&mitigation_profile0>;
 			qcom,lmh-dcvs = <&lmh_dcvs0>;
@@ -157,7 +157,7 @@
 			device_type = "cpu";
 			compatible = "arm,kryo";
 			reg = <0x0 0x100>;
-			clock-frequency = <1804800000>;
+			clock-frequency = <2208000000>;
 			enable-method = "psci";
 			qcom,limits-info = <&mitigation_profile1>;
 			qcom,lmh-dcvs = <&lmh_dcvs1>;
@@ -186,7 +186,7 @@
 			device_type = "cpu";
 			compatible = "arm,kryo";
 			reg = <0x0 0x101>;
-			clock-frequency = <1804800000>;
+			clock-frequency = <2208000000>;
 			enable-method = "psci";
 			qcom,limits-info = <&mitigation_profile2>;
 			qcom,lmh-dcvs = <&lmh_dcvs1>;
@@ -211,7 +211,7 @@
 			device_type = "cpu";
 			compatible = "arm,kryo";
 			reg = <0x0 0x102>;
-			clock-frequency = <1804800000>;
+			clock-frequency = <2208000000>;
 			enable-method = "psci";
 			qcom,limits-info = <&mitigation_profile3>;
 			qcom,lmh-dcvs = <&lmh_dcvs1>;
@@ -236,7 +236,7 @@
 			device_type = "cpu";
 			compatible = "arm,kryo";
 			reg = <0x0 0x103>;
-			clock-frequency = <1804800000>;
+			clock-frequency = <2208000000>;
 			enable-method = "psci";
 			qcom,limits-info = <&mitigation_profile4>;
 			qcom,lmh-dcvs = <&lmh_dcvs1>;
@@ -307,6 +307,8 @@
 					1401600 146
 					1536000 158
 					1612800 183
+					1747200 228
+					1843200 285
 				>;
 				idle-cost-data = <
 					20 16 12 8
@@ -318,6 +320,9 @@
 					1401600 485
 					1747200 857
 					1804800 883
+					1958400 1222
+					2150400 1592
+					2208000 1632
 				>;
 				idle-cost-data = <
 					100 80 60 40
@@ -331,6 +336,8 @@
 					1401600 9
 					1536000 9
 					1612800 10
+					1747200 11
+					1843200 13
 				>;
 				idle-cost-data = <
 					4 3 2 1
@@ -342,6 +349,9 @@
 					1401600 17
 					1747200 25
 					1804800 31
+					1958400 31
+					2150400 37
+					2208000 44
 				>;
 				idle-cost-data = <
 					4 3 2 1
@@ -1318,7 +1328,7 @@
 		qcom,core-dev-table =
 			< 902400 762 >,
 			< 1401600 2086 >,
-			< 1881600 3879 >;
+			< 1843200 3879 >;
 	};
 
 	devfreq_memlat_4: qcom,arm-memlat-mon-4 {
@@ -1329,7 +1339,7 @@
 			< 1113600 762 >,
 			< 1401600 3879 >,
 			< 2150400 5931 >,
-			< 2457600 6881 >;
+			< 2208000 6881 >;
 	};
 
 	devfreq_cpufreq: devfreq-cpufreq {
@@ -1338,13 +1348,13 @@
 			cpu-to-dev-map-0 =
 				< 633600 762 >,
 				< 1401600 1571 >,
-				< 1881600 2929 >;
+				< 1843200 2929 >;
 			cpu-to-dev-map-4 =
 				< 1113600 762 >,
 				< 1401600 2086 >,
 				< 1747200 2929 >,
 				< 2150400 3879 >,
-				< 2457600 6881 >;
+				< 2208000 6881 >;
 		};
 	};
 
@@ -1391,7 +1401,8 @@
 			<  1113600000 0x0404003a 0x052e002e 0x2 4 >,
 			<  1401600000 0x04040049 0x073a003a 0x2 5 >,
 			<  1536000000 0x04040050 0x08400040 0x2 6 >,
-			<  1612800000 0x04040054 0x09430043 0x2 7 >;
+			<  1747200000 0x0404005b 0x09480048 0x2 7 >,
+			<  1843200000 0x04040060 0x094c004c 0x3 8 >;
 
 		qcom,pwrcl-speedbin4-v0 =
 			<   300000000 0x0004000f 0x01200020 0x1 1 >,
@@ -1426,7 +1437,10 @@
 			<  1113600000 0x0404003a 0x052e002e 0x1 2 >,
 			<  1401600000 0x04040049 0x073a003a 0x2 3 >,
 			<  1747200000 0x0404005b 0x09480048 0x2 4 >,
-			<  1804800000 0x0404005e 0x094b004b 0x2 5 >;
+			<  1958400000 0x04040066 0x0a510051 0x2 5 >,
+			<  2150400000 0x04040070 0x0b590059 0x2 6 >,
+			<  2208000000 0x04040073 0x0b5c005c 0x3 7 >;
+
 
 		qcom,perfcl-speedbin4-v0 =
 			<   300000000 0x0004000f 0x01200020 0x1 1 >,
